### PR TITLE
Allow passing multiple metrics to combine in one graph

### PR DESCRIPTION
**Pull request by fishnix**
* allow passing multiple metrics as query params to be combined into one graph
* fix metrics period parsing (was defaulting to 300s before)
* fix start/end for space cost

### CODE REVIEW

File: s3cache/s3cache.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func (s *S3Cache) Save(ctx context.Context, key string, obj []byte) ([]byte, err
 }
 
 func (s *S3Cache) HashedKey(key string) string {
-	hasher := sha256.New()
+	hasher := sha512.New()
```
Comment (tenyo) [inline s3cache/s3cache.go]: 🔒

File: api/handlers_metrics.go [hunk 2]
```diff
@@ -65,25 +78,36 @@ func (s *server) MetricsGetImageHandler(w http.ResponseWriter, r *http.Request)
...
+		handleError(w, apierror.New(apierror.ErrBadRequest, "at least one metric is required", nil))
+		return
+	}
+	sort.Strings(metrics)
+
 	period := int64(300)
-	// 	// TODO p is a string, need int64
-	// if p, ok := vars["period"]; ok {
-	// 	period = p
-	// }
```
Comment (tenyo) [inline api/handlers_metrics.go]: 👀

File: api/handlers_metrics.go [hunk 3]
```diff
@@ -15,17 +17,27 @@ import (
...
+	if len(metrics) == 0 {
+		handleError(w, apierror.New(apierror.ErrBadRequest, "at least one metric is required", nil))
+		return
+	}
+
 	period := int64(300)
-	// if p, ok := vars["period"]; ok {
-	// 	// TODO p is a string, need int64
-	// 	period = p
-	// }
```
Comment (tenyo) [inline api/handlers_metrics.go]: 👀

File: api/handlers_metrics.go (hunk 2 above)
Comment (fishnix) [reply api/handlers_metrics.go]: ? 😕

File: api/handlers_metrics.go (hunk 3 above)
Comment (fishnix) [reply api/handlers_metrics.go]: ? 😕

File: api/server.go [hunk 4]
```diff
@@ -78,7 +78,7 @@ func NewServer(config common.Config) error {
 		return err
 	}
 
-	// Create a shared Cost Explorer session
+	// Create shared cost explorer sessions, cloudwatch sessions, and go-cache instances per account defained in the config
```
Comment (damnski) [inline api/server.go]: s/defained/defined/

**Comment by damnski** [comment]
Looks good besides minor change. :-)

**Pull request merged**